python/REDSHIFT_CLUSTER_PUBLIC_ACCESS_CHECK/REDSHIFT_CLUSTER_PUBLIC_ACCESS_CHECK.py

```diff
@@ -0,0 +1,343 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+"""
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+    REDSHIFT_CLUSTER_PUBLIC_ACCESS_CHECK
+Description:
+    Check whether the Amazon Redshift clusters are not publicly accessible. The rule is NON_COMPLIANT if the publiclyAccessible field is true in the cluster configuration item.
+Trigger:
+    Configuration Change on AWS::Redshift::Cluster
+Reports on:
+    AWS::Redshift::Cluster
+Rule Parameters:
+    None
+
+Scenarios:
+  Scenario: 1
+    Given: The publiclyAccessible field is true in the cluster configuration item.
+    Then: Return NON_COMPLIANT
+  Scenario: 2
+    Given: The publiclyAccessible field is false in the cluster configuration item.
+    Then: Return COMPLIANT
+"""
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::Redshift::Cluster'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    ###############################
+    # Add your custom logic here. #
+    ###############################
+    if not configuration_item['configuration']['publiclyAccessible']:
+        return build_evaluation_from_config_item(configuration_item, 'COMPLIANT')
+    return build_evaluation_from_config_item(configuration_item, 'NON_COMPLIANT', annotation='Amazon Redshift Cluster allows public access.')

```

Comment:
Annotation: 
This Amazon Redshift Cluster has the publiclyAccessible field set to True.